app/services/open_library.rb

```diff
@@ -0,0 +1,23 @@
+class OpenLibrary
+  include HTTParty
+  base_uri 'https://openlibrary.org'
+
+  def initialize(isbn)
+    @isbn = "ISBN:#{isbn}"
+    @options = { query: { bibkeys: "ISBN:#{isbn}", format: 'json', jscmd: 'data' } }
+  end
+
+  def books
+    process_response self.class.get('/api/books', @options)
+  end
+
+  def process_response(response)
+    result = { ISBN: @isbn }
+    response_body = JSON.parse(response.body)
```

Comment:
Only for general knowledge! (You not need change it) You can resolve this just like that...
```ruby
response_body = JSON.parse(response.body)
keys = %w[title subtitle number_of_pages authors]
keys.each do |key|
  result[key.to_sym] = response_body.dig(@isbn, key)
end
result
```
It's better? Not? Maybe? Yes? I don't know, it just I'm bored. 😄

Reply:
I think a better alternative is:

```rb
response_body =  JSON.parse(response.body)
response_body[@isbn]
  .slice(:title, :subtitle, :number_of_pages, :authors)
  .merge(ISBN: @isbn) 
```

But it doesn't make a lot of difference